principles/1_SecurityImplementation.md

```diff
@@ -0,0 +1,61 @@
+* MSTG-PLATFORM-3 The app does not export sensitive functionality via custom URL schemes, unless these mechanisms are properly protected.
+* MSTG-PLATFORM-4 The app does not export sensitive functionality through IPC facilities, unless these mechanisms are properly protected.
+* MSTG-PLATFORM-8 Object deserialization, if any, is implemented using safe serialization APIs.
+* MSTG-CODE-1 The app is signed and provisioned with a valid certificate, of which the private key is properly protected.
+* MSTG-CODE-2 The app has been built in release mode, with settings appropriate for a release build (e.g. non-debuggable).
+* MSTG-CODE-4 Debugging code and developer assistance code (e.g. test code, backdoors, hidden settings) have been removed. The app does not log verbose errors or debugging messages.
+* MSTG-CODE-5 All third party components used by the mobile app, such as libraries and frameworks, are identified, and checked for known vulnerabilities.
+* MSTG-CODE-6 The app catches and handles possible exceptions.
+* MSTG-CODE-8 In unmanaged code, memory is allocated, freed and used securely.
+* MSTG-CODE-9 Free security features offered by the toolchain, such as byte-code minification, stack protection, PIE support and automatic reference counting, are activated.
+* MSTG-STORAGE-1: System credential storage facilities need to be used to store sensitive data, such as PII, user credentials or cryptographic keys.
+* MSTG-STORAGE-2: No sensitive data should be stored outside of the app container or system credential storage facilities.
+* MSTG-STORAGE-3: No sensitive data is written to application logs.
+* MSTG-STORAGE-4: No sensitive data is shared with third parties unless it is a necessary part of the architecture.
+* MSTG-STORAGE-5: The keyboard cache is disabled on text inputs that process sensitive data.
+* MSTG-STORAGE-6: No sensitive data is exposed via IPC mechanisms.
+* MSTG-STORAGE-7: No sensitive data, such as passwords or pins, is exposed through the user interface.
+* MSTG-STORAGE-8: No sensitive data is included in backups generated by the mobile operating system.
+* MSTG-STORAGE-9: The app removes sensitive data from views when moved to the background.
+* MSTG-STORAGE-10 The app does not hold sensitive data in memory longer than necessary, and memory is cleared explicitly after use.
+
+
+## Safe Places
+
+### General
+
+* Whilst Safe Places deployment security best practices are the responsibility of the healthcare authority, developers shall make every effort to encourage healthcare authorities to deploy in a secure way, through documentation and technical means
+* Access to sensitive data shall be logged and made visible to users
+* Security and data access logs shall be immutable
+* Aggregated data published from Safe Places shall not be accessible in plain text to a layman user or malicious actor
```

Comment:
This might be sort of achievable with a Public Key Infrastructure, (PKI) but that was deemed beyond the scope  at present.
Also, since the app needs to be able to decide from the data it gets from a public URL, though it might not be "plain text" it also can't be completely encrypted as info about points of concern my understanding must be present and a malicious actor can look just like a non-malicious MoP user.

Reply:
Agreed.. these principles should be the goal though - and we will document anything that isn't achieved for MVP1.